### PR TITLE
adding weight column to lightgbm

### PR DESCRIPTION
**Pull request by imatiach-msft**
adding weight column to lightgbm + tests
resolves issue: https://github.com/Azure/mmlspark/issues/411

### CODE REVIEW

File: src/lightgbm/src/main/scala/LightGBMClassifier.scala [hunk 1]
```diff
@@ -54,8 +54,10 @@ class LightGBMClassifier(override val uid: String)
      */
     val actualNumClasses = getNumClasses(dataset)
     val networkParams = NetworkParams(getDefaultListenPort, inetAddress, port)
+    val optionalWeightCol = if (hasWeightCol) Some(getWeightCol) else None
```
Comment (mhamilton723) [inline src/lightgbm/src/main/scala/LightGBMClassifier.scala]:
```suggestion
    val optionalWeightCol = get(weightCol)
```

Or something like this

File: src/lightgbm/src/main/scala/LightGBMClassifier.scala [hunk 2]
```diff
@@ -54,8 +54,10 @@ class LightGBMClassifier(override val uid: String)
...
+
     val lightGBMBooster = df
-      .mapPartitions(TrainUtils.trainLightGBM(networkParams, getLabelCol, getFeaturesCol,
+      .mapPartitions(TrainUtils.trainLightGBM(networkParams, getLabelCol, getFeaturesCol, optionalWeightCol,
```
Comment (mhamilton723) [inline src/lightgbm/src/main/scala/LightGBMClassifier.scala]: Can probably put the get(weightCol) here if not used elsewhere

File: src/lightgbm/src/main/scala/LightGBMParams.scala [hunk 3]
```diff
@@ -111,4 +111,8 @@ trait LightGBMParams extends Wrappable with DefaultParamsWritable {
 
   def getModelString: String = $(modelString)
   def setModelString(value: String): this.type = set(modelString, value)
+
+  def hasWeightCol: Boolean = {
+    isDefined(weightCol) && $(weightCol).nonEmpty
```
Comment (mhamilton723) [inline src/lightgbm/src/main/scala/LightGBMParams.scala]: Unnecessary

Comment (imatiach-msft) [reply src/lightgbm/src/main/scala/LightGBMParams.scala]: removed

File: src/lightgbm/src/main/scala/LightGBMClassifier.scala (hunk 2 above)
Comment (imatiach-msft) [reply src/lightgbm/src/main/scala/LightGBMClassifier.scala]: done!

File: src/lightgbm/src/main/scala/LightGBMClassifier.scala (hunk 1 above)
Comment (imatiach-msft) [reply src/lightgbm/src/main/scala/LightGBMClassifier.scala]: done!

**Pull request merged**